/*
 * This file was automatically generated by EvoSuite
 * Tue Nov 15 14:56:38 GMT 2022
 */

package net.spy.memcached.ops;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.shaded.org.mockito.Mockito.*;
import static org.evosuite.runtime.EvoAssertions.*;
import net.spy.memcached.ops.MultiGetsOperationCallback;
import net.spy.memcached.ops.OperationCallback;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.ViolatedAssumptionAnswer;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class MultiGetsOperationCallback_ESTest extends MultiGetsOperationCallback_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      OperationCallback operationCallback0 = mock(OperationCallback.class, new ViolatedAssumptionAnswer());
      MultiGetsOperationCallback multiGetsOperationCallback0 = new MultiGetsOperationCallback(operationCallback0, (-1368));
      // Undeclared exception!
      try { 
        multiGetsOperationCallback0.gotData("v<jJ:b~cNCJD&/<aa", (-1368), (-1368), (byte[]) null);
        fail("Expecting exception: ClassCastException");
      
      } catch(ClassCastException e) {
         //
         // class net.spy.memcached.ops.OperationCallback$MockitoMock$1941587443 cannot be cast to class net.spy.memcached.ops.GetsOperation$Callback (net.spy.memcached.ops.OperationCallback$MockitoMock$1941587443 and net.spy.memcached.ops.GetsOperation$Callback are in unnamed module of loader org.evosuite.instrumentation.InstrumentingClassLoader @78e20115)
         //
         verifyException("net.spy.memcached.ops.MultiGetsOperationCallback", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      MultiGetsOperationCallback multiGetsOperationCallback0 = new MultiGetsOperationCallback((OperationCallback) null, (-345));
      // Undeclared exception!
      try { 
        multiGetsOperationCallback0.gotData("<j}J$0FLd;CzK`'~2", (-345), (-345), (byte[]) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("net.spy.memcached.ops.MultiGetsOperationCallback", e);
      }
  }
}
